infrastructure/aws/main.tf

```diff
@@ -0,0 +1,155 @@
+data "aws_availability_zones" "available" {}
+
+resource "aws_vpc" "eks_vpc" {
+  cidr_block = var.vpc_cidr
+
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+
+  tags = {
+    Name = "${var.cluster_name}-vpc"
+  }
+}
+
+resource "aws_subnet" "eks_subnets" {
+  count                   = length(var.availability_zones)
+  vpc_id                  = aws_vpc.eks_vpc.id
+  cidr_block              = cidrsubnet(var.vpc_cidr, 8, count.index)
+  availability_zone       = var.availability_zones[count.index]
+  map_public_ip_on_launch = true
+
+  tags = {
+    Name                                        = "${var.cluster_name}-subnet-${count.index + 1}"
+    "kubernetes.io/cluster/${var.cluster_name}" = "shared"
+  }
+}
+
+resource "aws_internet_gateway" "eks_igw" {
```

Comment:
I kinda want to spell out `eks_internet_gateway` here personally.

Reply:
good call, I hate acronyms, fixing